public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -192,13 +195,72 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
     ).finally(() => setIsSpansTableDataLoading(false));
   };
 
+  // For application_analytics
   useEffect(() => {
-    fetchData();
+    if (!props.payloadData) {
+      fetchData();
+    }
   }, [tableParams, props.DSL]);
 
   useEffect(() => {
-    if (props.setTotal) props.setTotal(total);
-  }, [total]);
+    if (!props.payloadData) {
+      console.warn('No payloadData provided in SpanDetailTable');
+      return;
+    }
+    try {
+      const hitsArray = parseHits(props.payloadData);
+
+      // Map each hit to its _source
+      let spans = hitsArray.map((hit: any) => hit._source);
+
+      // Apply filters passed as a prop.
+      if (props.filters.length > 0) {
+        spans = spans.filter((span: any) => {
+          return props.filters.every(({ field, value }) => {
+            return span[field] === value;
+          });
+        });
+      }
+
+      spans = applySorting(spans);
+
+      setItems(spans);
+      setTotal(spans.length);
+    } catch (error) {
+      console.error('Error parsing payloadData in SpanDetailTable:', error);
+    } finally {
+      setIsSpansTableDataLoading(false);
+    }
+  }, [props.payloadData, props.DSL, props.filters, tableParams]);
+
+  const applySorting = (spans: Span[]) => {
+    if (tableParams.sortingColumns.length > 0) {
```

Comment:
We should move this condition check outside of this function. The caller should handle sorting and ensure the array is not empty before invoking this function to perform the sorting.

Reply:
Updated, moved the check to the useEffect calling it.